src/runtime/schema/schema.ts

```diff
@@ -98,6 +99,8 @@ export function create(): SchemaInternal {
             )
           }
 
+          const contextExtractionPromise = runAddToContextExtractorAsWorkerIfPossible(devModeLayout.data)
```

Comment:
Why not run this via watcher plugin? E.g. startOrRestart hook?

Reply:
Should we then move the nexus typegen out of here as well?
Because it doesn't make sense to me that we treat the context type extraction different than typegen in general, or does it?

Reply:
That sounds great

Reply:
Could be a follow up refactor/PR!

Reply:
Actually, doing this is gonna require us to always run the app twice. One to "only generate types", and one to run the app. Do we really want that? :/